src/many-ledger/src/storage.rs

```diff
@@ -17,24 +17,18 @@ const HEIGHT_TXID_SHIFT: u64 = 32;
 
 /// Minimum number of bytes in a transaction ID. If a transaction ID would be
 /// larger than this, panics.
-const TRANSACTION_ID_KEY_SIZE: usize = 32;
+const TRANSACTION_ID_KEY_SIZE_IN_BYTES: usize = 32;
 
 /// Returns the key for the persistent kv-store.
 pub(crate) fn key_for_account(id: &Identity, symbol: &Symbol) -> Vec<u8> {
     format!("/balances/{}/{}", id, symbol).into_bytes()
 }
 
 /// Returns the storage key for a transaction in the kv-store.
-pub(crate) fn key_for_transaction(id: TransactionId) -> Vec<u8> {
-    let mut id: Vec<u8> = id.0.into();
-    id.extend(
-        std::iter::repeat(0).take(
-            TRANSACTION_ID_KEY_SIZE
-                .checked_sub(id.len())
-                .expect("Transaction ID larger than max size."),
-        ),
-    );
-    vec![TRANSACTIONS_ROOT.to_vec(), id].concat()
+pub(super) fn key_for_transaction(id: TransactionId) -> Vec<u8> {
+    let mut exp_id = [0u8; TRANSACTION_ID_KEY_SIZE_IN_BYTES];
+    exp_id[(TRANSACTION_ID_KEY_SIZE_IN_BYTES - id.0.len())..].copy_from_slice(&id.0);
```

Comment:
I **think** this can be abused via the LedgerStorage iterator right now, i.e. this can panic with an overflow.

Ex:
```
    storage
        .iter(
            CborRange {
                start: Bound::Unbounded,
                end: Bound::Included(TransactionId::from(vec![1u8; 33])),
            },
            SortOrder::Ascending,
        )
        .all(|_| true);
```

WDYT?

Reply:
Yes, we know TransactionId with greater than 32 bytes will panic.

Reply:
I can't return a result. This needs to either panic or succeeds. Now there can be user entry so what I'll do is cut it instead of panicking.